From efb7058cb1ecc9769451ccc5911e7aa9ae06b903 Mon Sep 17 00:00:00 2001
From: aryamali <quic_aryamali@quicinc.com>
Date: Tue, 1 Apr 2025 15:55:25 -0700
Subject: [PATCH] odhcpd: removing IFF_RUNNING check

Fix bug where odhcpd does not register br-lan when no clients are attached

Signed-off-by: aryamali <quic_aryamali@quicinc.com>
---
 src/config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/config.c b/src/config.c
index 1cfc668..5f9f3c0 100644
--- a/src/config.c
+++ b/src/config.c
@@ -1413,7 +1413,7 @@ void odhcpd_reload(void)
 
 
 	avl_for_each_element_safe(&interfaces, i, avl, tmp) {
-		if (i->inuse && i->ifflags & IFF_RUNNING) {
+		if (i->inuse) {
 			/* Resolve hybrid mode */
 			if (i->dhcpv6 == MODE_HYBRID)
 				i->dhcpv6 = (master && master->dhcpv6 == MODE_RELAY) ?
-- 
2.51.0.windows.1

